CodeToReview.java

```diff
@@ -0,0 +1,250 @@
+                        winner = true;
+                        System.out.println("DRAW!");
+                        Thread.sleep(3000);
+                        break;
+                    } else {
+                        PrintBoard(a1,a2,a3,a4,a5,a6,a7,a8,a9);
+                        System.out.printf("Player O's turn: ");
+                        inO = input.nextInt();
+                        break;
+                    }
+                case "X":
+                    winner = true;
+                    System.out.println("X HAS WON!!!");
+                    Thread.sleep(3000);
+                    break;
+                case "O":
+                    winner = true;
+                    System.out.println("O HAS WON!!!");
+                    Thread.sleep(3000);
+                    break;
+                default:
+                    break;
+            }
+            if (winner) {
+                break;
+            }
+        }
+    }
+
+    public static void PrintBoard(String f1, String f2, String f3, String f4, String f5, String f6, String f7, String f8, String f9) {
+        System.out.printf("   %s | %s | %s \n",f1,f2,f3);
+        System.out.printf("  ----------- \n");
+        System.out.printf("   %s | %s | %s \n",f4,f5,f6);
+        System.out.printf("  ----------- \n");
+        System.out.printf("   %s | %s | %s \n",f7,f8,f9);
+        System.out.printf(" \n");
+    }
+
+    public static String Winners(String f1, String f2, String f3, String f4, String f5, String f6, String f7, String f8, String f9) {
+        if (f1 == f2 && f2 == f3 && f3 == "O") {
+            return "O";
+        } else if (f4 == f5 && f5 == f6 && f6 == "O") {
+            return "O";
+        } else if (f7 == f8 && f8 == f9 && f9 == "O") {
+            return "O";
+        } else if (f1 == f4 && f4 == f7 && f7 == "O") {
+            return "O";
+        } else if (f2 == f5 && f5 == f8 && f8 == "O") {
+            return "O";
+        } else if (f3 == f6 && f6 == f9 && f9 == "O") {
+            return "O";
+        } else if (f1 == f5 && f5 == f9 && f9 == "O") {
+            return "O";
+        } else if (f3 == f5 && f5 == f7 && f7 == "O") {
+            return "O";
+        } else if (f1 == f2 && f2 == f3 && f3 == "X") {
+            return "X";
+        } else if (f4 == f5 && f5 == f6 && f6 == "X") {
+            return "X";
+        } else if (f7 == f8 && f8 == f9 && f9 == "X") {
+            return "X";
+        } else if (f1 == f4 && f4 == f7 && f7 == "X") {
+            return "X";
+        } else if (f2 == f5 && f5 == f8 && f8 == "X") {
+            return "X";
+        } else if (f3 == f6 && f6 == f9 && f9 == "X") {
+            return "X";
+        } else if (f1 == f5 && f5 == f9 && f9 == "X") {
+            return "X";
+        } else if (f3 == f5 && f5 == f7 && f7 == "X") {
+            return "X";
+        } else {
+            return "NONE";
```

Comment:
Unreadable. Should write a generalized function to find the winner, instead of brute forcing all winning situations.